src/Infrastructure/Event/ApplicationReceiverSetupListener.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+/**
+ * Copyright © 2019-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerSdk\Sdk\Infrastructure\Event;
+
+use SprykerSdk\Sdk\Infrastructure\Service\ValueReceiver\ApiInteractionProcessor;
+use SprykerSdk\Sdk\Infrastructure\Service\ValueReceiver\CliInteractionProcessor;
+use SprykerSdk\Sdk\Infrastructure\Service\ValueReceiver\InteractionProcessorInjectorInterface;
+use Symfony\Component\Console\Event\ConsoleCommandEvent;
+use Symfony\Component\Console\Helper\DebugFormatterHelper;
+use Symfony\Component\Console\Helper\FormatterHelper;
+use Symfony\Component\Console\Helper\HelperSet;
+use Symfony\Component\Console\Helper\ProcessHelper;
+use Symfony\Component\Console\Helper\QuestionHelper;
+use Symfony\Component\Console\Output\BufferedOutput;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+
+class ApplicationReceiverSetupListener
+{
+    /**
+     * @var iterable<\SprykerSdk\Sdk\Infrastructure\Event\InjectorInterface>
+     */
+    protected iterable $inputOutputConnectors;
+
+    /**
+     * @var \SprykerSdk\Sdk\Infrastructure\Service\ValueReceiver\ApiInteractionProcessor
+     */
+    protected ApiInteractionProcessor $apiInteractionProcessor;
+
+    /**
+     * @var \SprykerSdk\Sdk\Infrastructure\Service\ValueReceiver\CliInteractionProcessor
+     */
+    protected CliInteractionProcessor $cliInteractionProcessor;
+
+    /**
+     * @param iterable $inputOutputConnectors
+     * @param \SprykerSdk\Sdk\Infrastructure\Service\ValueReceiver\ApiInteractionProcessor $apiInteractionProcessor
+     * @param \SprykerSdk\Sdk\Infrastructure\Service\ValueReceiver\CliInteractionProcessor $cliInteractionProcessor
+     */
+    public function __construct(
+        iterable $inputOutputConnectors,
+        ApiInteractionProcessor $apiInteractionProcessor,
+        CliInteractionProcessor $cliInteractionProcessor
+    ) {
+        $this->inputOutputConnectors = $inputOutputConnectors;
+        $this->apiInteractionProcessor = $apiInteractionProcessor;
+        $this->cliInteractionProcessor = $cliInteractionProcessor;
+    }
+
+    /**
+     * @param \Symfony\Component\Console\Event\ConsoleCommandEvent $event
+     *
+     * @return void
+     */
+    public function beforeConsoleCommand(ConsoleCommandEvent $event): void
+    {
+        if (!$event->getCommand()) {
+            return;
+        }
+
+        foreach ($this->inputOutputConnectors as $inputOutputConnector) {
+            if ($inputOutputConnector instanceof InputInjectorInterface) {
```

Comment:
In this case, instead of using multiple interfaces, it is better to add method `inject` which accepts `specific object`. And each concrete injector will decide what it should do.

The same for the `onKernelRequest`

Reply:
We will refactor it later. We probably will rewrite some part. needs to see all troubles

Reply:
I'll make a ticket then.

Reply:
https://spryker.atlassian.net/browse/SDK-1091